src/context.rs

```diff
@@ -979,6 +988,11 @@ impl<'context> WidgetContext<'context> {
         if let Some(mode) = overridden_theme_mode {
             context.cache.theme_mode = mode.get_tracking_redraw(&context);
         }
+        if let Some(locale) = overridden_locale {
+            context.locale = locale
+        }
+        context.locale.invalidate_when_changed(&context);
+        context.locale.redraw_when_changed(&context);
```

Comment:
Each source only notifies any given widget of an invalidation or a redraw. This is because invalidating a widget causes a redraw automatically, since invalidation causes widgets to re-layout, which must be redrawn to be shown. So in short, for any given dynamic, either redraw or invalidation tracking should be used, not both. Either the dynamic potentially affects layout and should invalidate when changed, or it should only redraw when changed.